### PR TITLE
SVA-80 | Filter upcoming events

### PR DESCRIPTION
**Pull request by joefhall**
- [Description](#description)
  - [Type of change](#type-of-change)
- [Checklist](#checklist)

Events upcoming date filter

# Description

- Add quick search filtering for upcoming events - today, this week, this month
- Also moved the calendar date picker into its own component and finished wiring it up for upcoming events so can filter that way too
- Instead of using a separate container for search results like we've done for projects, realised the search results screen is essentially very similar to the main event listings screen, so have just made the EventsContainer more flexible to handle both
- A little bit of tidying/improving some of EventsContainer and related components
- Altered how the API returns event durations -- as number of minutes (e.g. 60) instead of string (e.g. '1 hour').  This machine-readable format is more useful for calculating event end times.

Fixes [SVA-80](https://sta2020.atlassian.net/jira/software/projects/SVA/boards/145?selectedIssue=SVA-80)

## Type of change

- [x] New Feature (non-breaking change which adds functionality)

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation (if required)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/src/Containers/EventsContainer.tsx [hunk 1]
```diff
@@ -23,29 +31,109 @@ const HorizontalLine = styled.View`
...
+  const EventList: FC<EventProps> = ({ data }) => {
+    return (
+      <SafeArea>
+        <TopOfApp />
+        <EventSearch />
+        <EventOptions selected={EventsRange.Upcoming} />
+
+        {/* If the user has done a quick search for upcoming events, show those
+            quick search buttons so they can amend their quick search if they want,
+            without having to go back to the search screen */}
```
Comment (joefhall) [inline app/src/Containers/EventsContainer.tsx:99]: From the current Figma designs, it looks like these quick search buttons should also appear on the results screen when they've been used for a search

File: api/__test-data__/events.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 const { faker } = require('@faker-js/faker');
 
 const eventTypes = ['Internal', 'External'];
-const durations = ['1 hour', '2 hours', '4 hours 30 minutes', '1 day'];
+const durations = [60, 120, 360, 480];
```
Comment (DougieWougie) [inline api/__test-data__/events.js:4]: Should we add a comment explaining these - it's obvious now but might appear as magic numbers later!

Comment (joefhall) [reply api/__test-data__/events.js]: Good shout, thanks -- have made that clearer now 👍

**Review by DougieWougie — APPROVED** [review]
Great work @joefhall !

**Comment by M4RC1N76** [comment]
I have run unit test with pass. Also checked the Volunteer App Figma design for upcoming events and familiarise myself with (api/__test-data__/events.js) comments regarding reason for change in the values. Grand job Joe.

**Review by M4RC1N76 — APPROVED** [review]

**Comment by joefhall** [comment]
Thanks @DougieWougie and @M4RC1N76! 👍

**Pull request merged**